src/physics/em/PhotoelectricMicroXsCalculator.i.hh

```diff
@@ -0,0 +1,78 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file PhotoelectricMicroXsCalculator.i.hh
+//---------------------------------------------------------------------------//
+
+#include "base/Algorithms.hh"
+#include "MockXsCalculator.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with shared and state data.
+ */
+CELER_FUNCTION PhotoelectricMicroXsCalculator::PhotoelectricMicroXsCalculator(
+    const PhotoelectricInteractorPointers& shared,
+    const LivermoreParamsPointers&         data,
+    const ParticleTrackView&               particle)
+    : shared_(shared), data_(data), inc_energy_(particle.energy().value())
+{
+    REQUIRE(particle.def_id() == shared_.gamma_id);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Compute cross section
+ */
+CELER_FUNCTION
+real_type PhotoelectricMicroXsCalculator::operator()(ElementDefId el_id) const
+{
+    REQUIRE(el_id);
+    const LivermoreElement& el = data_.elements[el_id.get()];
+
+    // In Geant4, if the incident gamma energy is below the lowest binding
+    // energy, it is set to the binding energy so that the photoelectric cross
+    // section is constant rather than zero for low energy gammas.
+    MevEnergy energy
+        = max(inc_energy_, el.shells[el.shells.size() - 1].binding_energy);
```

Comment:
I think `el.shells.back()` is more readable than `el.shells[el.shells.size() - 1]`. There's another case below that could use the change, and another place could be `el.shells.front()`.

Reply:
Yep it is!